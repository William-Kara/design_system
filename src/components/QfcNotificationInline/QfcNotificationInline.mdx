import { ArgTypes, Meta } from '@storybook/blocks';
import QfcNotificationInline from '@/components/QfcNotificationInline/QfcNotificationInline.vue';

<Meta title="Modules/Notifications/NotificationInline" />

# Definition & Usage

## Props

The following props are available for the QfcNotificationInline component:

<ArgTypes of={QfcNotificationInline} />

To use QfcNotificationInline, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcBox.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcNotificationInline.scss';
```

```html
<template>
  <QfcNotificationInline
    time-ago="1 hour ago"
    :read="false"
    @click="click()"
    button="Validate"
    description="Description content"
    tag="Tag filter"
  />
</template>
```
