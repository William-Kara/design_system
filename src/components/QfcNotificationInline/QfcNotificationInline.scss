.qfc-notification-inline {
  display: flex;
  align-items: center;
  justify-content: space-between;

  &__content {
    display: flex;
    flex-direction: column;
    gap: $q-generic-space-2;
    font-size: $fs-m;
    font-weight: $medium;

    &__metadata {
      gap: $q-gap-m;
      display: flex;
      align-items: center;
      flex-shrink: 0;

      .time {
        color: #bababa;
        flex-shrink: 0;
      }
      .tag-label {
        flex-shrink: 0;
        background: #232323 no-repeat padding-box;
        border-radius: 4px;
      }
    }
    &__description {
    }
  }
  &__button {
  }
}
