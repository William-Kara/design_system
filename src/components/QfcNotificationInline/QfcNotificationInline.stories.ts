import NotificationInline from './QfcNotificationInline.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof NotificationInline> = {
  component: NotificationInline,
  title: 'Modules/Notifications/NotificationInline',
  argTypes: {
    timeAgo: String,
    read: Boolean,
    button: String,
    description: String,
    tag: String,
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    timeAgo: '1 hour ago',
    read: false,
    button: 'Secondary button',
    description: 'Information requiring notification',
    tag: 'Tag filter',
  },
};
