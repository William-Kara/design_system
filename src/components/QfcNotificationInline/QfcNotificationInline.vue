<template>
  <QfcNotificationBox :type="readStatus">
    <div class="qfc-notification-inline">
      <div class="qfc-notification-inline__content">
        <div class="qfc-notification-inline__content__metadata">
          <span class="time">{{ timeAgo }}</span>
          <span class="tag-label px-3">{{ tag }}</span>
        </div>
        <span class="qfc-notification-inline__content__description">{{ description }}</span>
      </div>

      <QfcButton
        class="qfc-notification-inline__button"
        :label="button.label"
        :secondary="true"
        :transparent="true"
      />
    </div>
  </QfcNotificationBox>
</template>

<script lang="ts" setup>
import QfcNotificationBox from '@/components/QfcNotificationBox/QfcNotificationBox.vue';
import { computed } from 'vue';
import QfcButton from '@/components/QfcButton/QfcButton.vue';

interface Props {
  read: boolean;
  button: {
    label: string;
  };
  description: string;
  tag: string;
  timeAgo: string;
}
const props = defineProps<Props>();

const readStatus = computed(() => {
  return props.read ? 'none' : 'info';
});
</script>

<style lang="scss" scoped src="./QfcNotificationInline.scss" />
