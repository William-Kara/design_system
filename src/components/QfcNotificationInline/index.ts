import QfcNotificationInline from './QfcNotificationInline.vue';

export { QfcNotificationInline };
export default QfcNotificationInline;
