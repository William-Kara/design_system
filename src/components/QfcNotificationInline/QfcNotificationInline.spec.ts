import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcNotificationInline from './QfcNotificationInline.vue';

let wrapper: any = null;

const propsValues = {
  timeAgo: '1 hour ago',
  read: false,
  button: 'Secondary button',
  description: 'Information requiring notification',
  tag: 'Tag filter',
};

beforeEach(() => {
  wrapper = shallowMount(QfcNotificationInline, {
    propsData: propsValues,
  });
});

describe('QfcNotificationInline Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
