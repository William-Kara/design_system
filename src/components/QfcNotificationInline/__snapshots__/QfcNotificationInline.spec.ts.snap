// Vitest Snapshot v1

exports[`QfcNotificationInline Component > should render correctly 1`] = `"<qfc-notification-box-stub data-v-f86edb4b=\\"\\" type=\\"info\\"></qfc-notification-box-stub>"`;
